docs/source/en/model_doc/whisper.md

```diff
@@ -35,6 +35,85 @@ Tips:
 
+## Inference
+
+Here is a step-by-step guide to transcribing an audio sample using a pre-trained Whisper model:
+
+```python
+>>> import torchaudio
+>>> from transformers import WhisperProcessor, WhisperForConditionalGeneration
+
+>>> # Select an audio file:
+>>> audio_path = "https://huggingface.co/datasets/sanchit-gandhi/librispeech_long/resolve/main/audio.wav"
+
+>>> # Load the Whisper model in Hugging Face format:
+>>> processor = WhisperProcessor.from_pretrained("openai/whisper-tiny.en")
+>>> model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny.en")
+>>> model.config.forced_decoder_ids = None
+
+>>> # Select an audio file:
+>>> waveform, sampling_rate = torchaudio.load(audio_path)
+
+>>> # Use the model and processor to transcribe the audio:
+>>> input_features = processor(
+...     waveform.squeeze().numpy(), sampling_rate=sampling_rate, return_tensors="pt"
+... ).input_features
+
+>>> # Generate token ids
+>>> predicted_ids = model.generate(input_features)
+
+>>> # Decode token ids to text
+>>> transcription = processor.batch_decode(predicted_ids, skip_special_tokens=True)
+
+>>> transcription[0]
+' Chapter 16.'
+```
+
+## Format Conversion
+
+For users with models in the original OpenAI format who wish to utilize them with the Hugging Face library, a conversion script is provided. The example below demonstrates how to transform Whisper models from OpenAI to Hugging Face format:
+
+```bash
+# Change to the whisper directory where the script resides:
+cd src/transformers/models/whisper/
+# Converts the model from OpenAI to Hugging Face format:
+convert_openai_to_hf.py \
+  --checkpoint_path tiny \
+  --pytorch_dump_folder_path whisper-tiny-hf
+```
+
+For those more comfortable working directly in Python, the conversion can also be achieved with the code snippet below:
+
+```python
+>>> from transformers.models.whisper.convert_openai_to_hf import convert_openai_whisper_to_tfms
+>>> convert_openai_whisper_to_tfms("tiny.en", "whisper-tiny.en-hf")  # doctest: +IGNORE_RESULT
+```
+
+Now can test it by doing inference with an audio file:
+
+```python
+>>> # Load the newly converted model:
+>>> processor = WhisperProcessor.from_pretrained("openai/whisper-tiny.en")
+>>> model = WhisperForConditionalGeneration.from_pretrained("whisper-tiny.en-hf")
+>>> model.config.forced_decoder_ids = None
+
+>>> # Select an audio file:
+>>> waveform, sampling_rate = torchaudio.load(audio_path)
+
+>>> # Use the model and processor to transcribe the audio:
+>>> input_features = processor(
+...     waveform.squeeze().numpy(), sampling_rate=sampling_rate, return_tensors="pt"
+... ).input_features
+
+>>> # Transcribe the example:
+>>> predicted_ids = model.generate(input_features)
+>>> transcription = processor.batch_decode(predicted_ids, skip_special_tokens=True)
+
+>>> transcription[0]
+' Chapter 16. I might have told you of the beginning of this liaison in a few lines'
```

Comment:
We get a different result here to the one we got before (`Chapter 16` only)? Is one of them incorrect?

Reply:
I do not know why, but the transcriptions in HF and OpenAI do not match using this audio. To further explore this, I’ve created [a Kaggle notebook](https://www.kaggle.com/code/xabierdezuazo/hugging-face-vs-openai-whisper-models-inference) and transcribed the audio here using `Tiny.en` from both libraries. This also includes what happens when we convert between formats using the script here (`convert_openai_to_hf.py`) and the one in #26854 (`convert_hf_to_openai.py`).

Here are the transcription results:

| Model           | Transcription[:30]                    | SHA256 Prefix |
|-----------------|----------------------------------|---------------|
| OpenAI          | `" Chapter 16 I might have told "` | 80e1a202      |
| Hugging Face    | `" Chapter 16."`                   | c9f30e2e      |
| OpenAI -&gt; HF | `" Chapter 16. I might have told"` | 090c43de      |
| HF -&gt; OpenAI | `" Chapter 16 I might have told "` | 80e1a202      |

From the results, transcriptions using the OpenAI library are consistent, even after conversion from the HF model, suggesting the conversion is accurate. However, the transcriptions using the HF library, both with your model in  [https://huggingface.co/openai/whisper-tiny.en](https://huggingface.co/openai/whisper-tiny.en), and after converting the original `tiny.en` from OpenAI, differ.

I ignore the reason for this. Maybe some post-processing step. I have also not tested other model sizes or audios.

Should I open a new ticket with this?